src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -96,9 +99,13 @@ private void processZipFiles(CloudBlobContainer container)
 
         ServiceBusHelper serviceBusHelper = serviceBusHelper();
 
-        for (ListBlobItem blobItem : container.listBlobs()) {
-            String zipFilename = FilenameUtils.getName(blobItem.getUri().toString());
-
+        // Randomise iteration order to minimise lease acquire contention
+        List<String> zipFilenames = new ArrayList<>();
+        container.listBlobs().forEach(
+            b -> zipFilenames.add(FilenameUtils.getName(b.getUri().toString()))
+        );
```

Comment:
perhaps you could just map the blobs?
```
List<String> zipFilenames =
  container
    .listBlobs()
    .stream()
    .map(b -> FilenameUtils.getName(b.getUri().toString()))
    .collect(toList())
```

Reply:
There are no guarantees this way on what list can be returned and I want an implementation of the RandomAccess interface as it makes shuffling more efficient. Passing a Supplier to collect() would work, but I decided to go "traditional".

Reply:
may be worth adding a comment on this then before someone who didn't see this PR comes up with an idea of refactoring it in a couple of weeks ;)